src/components/Storage/api/useStorageConfig.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useSelector } from 'react-redux';
+
+import { EmulatorConfig } from '../../../store/config';
+import { getStorageConfigResult } from '../../../store/config/selectors';
+import { hasError } from '../../../store/utils';
+
+// Suspense aware wrapper.
+export function useStorageConfig(): EmulatorConfig {
+  const config = useSelector(getStorageConfigResult);
+
+  if (!config) {
+    throw Promise.resolve();
```

Comment:
This will probably just cause repeated rerenders until storage config is loaded. Worse, if you are on the storage tab and then storage emulator stops (e.g. across restart), this even have potential to go in infinite loop.

While I'm also eager to see this solved, please wait until we have go/fire-emulator-concurrent before we jump into this. For this PR, please just copy the pattern for e.g. Firestore w.r.t config.

Reply:
Just to clarify, there's no need to rewrite the whole thing -- we just need to remove suspension from this hook and just get the fields from `useSelector`. Nested child components can still use this hook (and return type can be changed to not-nullable since we already got that deep in). Just need to change the route component (top level) a bit to resemble Firestore for loading / error states.